Sources/Kitura/Router.swift

```diff
@@ -21,14 +21,36 @@ import KituraTemplateEngine
 
 // MARK Router
 
-/// The `Router` class provides the external interface for the routing of requests to
-/// the appropriate code for handling. This includes:
+/// `Router` provides the external interface for routing requests to
+/// the appropriate code to handle them. This includes:
 ///
-///   - Routing requests to closures with the signature of `RouterHandler`
+///   - Routing requests to closures of type `RouterHandler`
 ///   - Routing requests to the handle function of classes that implement the
 ///    `RouterMiddleware` protocol.
-///   - Routing the request to a template engine to generate the appropriate output.
+///   - Routing requests to a template engine to generate the appropriate output.
 ///   - Serving the landing page when someone makes an HTTP request with a path of slash (/).
+///
+/// When initialising a `Router`, `mergeParameters` allows you to control whether
+/// the router will be able to access parameters matched in its parent router. For instance, in the example below
+/// if `mergeParameters` is set to `true`, `GET /Hello/Alex` will return "Hello Alex", but if set to `false`
+/// the `greeting` parameter will not be accessible and it will return just " Alex".
+/// ```swift
+/// let router = Router()
+/// let userRouter = Router(mergeParameters: true)
+///
+/// router.get("/:greeting") request, response, _ in
+///   let greeting = request.parameters["greeting"] ?? ""
+///   try response.send("\(greeting)").end()
+/// }
+///
+/// userRouter.get("/:user") { request, response, _ in
+///   let user = request.parameters["user"] ?? ""
+///   let greeting = request.parameters["greeting"] ?? ""
+///   try response.send("\(greeting) \(user)").end()
+/// }
+///
+/// helloRouter.all("/:greeting", middleware: userRouter)
```

Comment:
It's not the most common usage, but it took me a while to work out exactly what `mergeParameters` actually did so thought I should create a code example for it. Put it at the top of the file as it's quite a chunky example to include further down and wasn't sure where else it could go! Agree with what you're saying about it potentially being confusing having it as the main example.. @tunniclm had some ideas about other things that should be highlighted here, for example the usage of next(), so can think about adding things in a future PR.